src/bentoml/_internal/io_descriptors/base.py

```diff
@@ -175,18 +175,29 @@ def input_type(self) -> InputType:
     async def from_http_request(self, request: Request) -> IOType:
         raise NotImplementedError
 
-    @abstractmethod
     async def to_http_response(
         self, obj: IOType, ctx: Context | None = None
     ) -> Response:
-        raise NotImplementedError
+        return self.sync_to_http_response(obj, ctx)
 
     @abstractmethod
-    async def from_proto(self, field: t.Any) -> IOType:
+    def sync_to_http_response(
+        self, obj: IOType, ctx: Context | None = None
+    ) -> Response:
```

Comment:
I think we unfortunately can't add new abstract methods or change this base class as it's public and we support defining your own IO descriptors; sync / async methods will be included in the next incarnation, I think.

Reply:
(This is also what's causing the test failure, FWIW)